src/providers/google/gdrive-document.ts

```diff
@@ -0,0 +1,207 @@
+import BaseSyncHandler from "../BaseSyncHandler";
+import CONFIG from "../../config";
+import { SyncProviderLogEvent, SyncProviderLogLevel } from '../../interfaces'
+import {
+    SyncResponse,
+    SyncHandlerPosition,
+    SyncHandlerStatus,
+} from "../../interfaces";
+import { SchemaDocument } from "../../schemas";
+import { google, drive_v3 } from "googleapis";
+import { OAuth2Client } from 'google-auth-library';
+import { GaxiosResponse } from "gaxios";
+import { GoogleDriveHelpers } from "./helpers";
+
+const _ = require("lodash");
+
+export default class GoogleDriveDocument extends BaseSyncHandler {
+
+    public getName(): string {
+        return "google-drive-documents";
+    }
+
+    public getSchemaUri(): string {
+        return CONFIG.verida.schemas.DOCUMENT;
+    }
+
+    public getProviderApplicationUrl(): string {
+        return "https://drive.google.com";
+    }
+    
+    public getGoogleAuth(): OAuth2Client {
+        const TOKEN = {
+            access_token: this.connection.accessToken,
+            refresh_token: this.connection.refreshToken,
+            scope: "https://www.googleapis.com/auth/drive.readonly",
+            token_type: "Bearer",
+        };
+
+        const redirectUrl = "";
+
+        const oAuth2Client = new google.auth.OAuth2(
+            this.config.clientId,
+            this.config.clientSecret,
+            redirectUrl
+        );
+
+        oAuth2Client.setCredentials(TOKEN);
+
+        return oAuth2Client;
+    }
+
+    public getGoogleDrive(): drive_v3.Drive {
+        
+        const auth = this.getGoogleAuth();
+
+        const drive = google.drive({ version: "v3", auth: auth });
+        return drive;
+    }
+
+    public async _sync(
```

Comment:
Update `_sync()` to use the new `ItemsRangeTracker` strategy.

Reply:
> Update `_sync()` to use the new `ItemsRangeTracker` strategy.

Done